test/Test.ts

```diff
@@ -2,21 +2,202 @@ import { describe } from 'mocha';
+          const balanceVoter2Before = await sarco.balanceOf(voter2.address)
+
+          await collection.connect(voter1).claim()
+          await collection.connect(voter2).claim()
+
+          // confirm claimed SARCO has reached voters' addresses
+          expect(await sarco.balanceOf(voter1.address)).to.equal(balanceVoter1Before.add(voterReward))
+          expect(await sarco.balanceOf(voter2.address)).to.equal(balanceVoter2Before.add(voterReward))
+        })
+
+        it("should reset the voters internal accounting mapping balanceOf[]", async () => {
+          await collection.connect(voter1).claim()
+          await collection.connect(voter2).claim()
+
+          // check internal accounting clears balances of voters once they have claimed rewards
+          expect(await collection.balanceOf(voter1.address)).to.equal(zero)
+          expect(await collection.balanceOf(voter1.address)).to.equal(zero)
+        })
+
+        it("should reset total claimable tokens by voters once all have been claimed", async () => {
+          await collection.connect(voter1).claim()
+          await collection.connect(voter2).claim()
+
+          // check internal accounting clears total claimable tokens by voters
+          expect(await collection.claimableByVoters()).to.equal(zero)
+        })
+
+        it("emit Claim", async () => {
+          expect(await collection.connect(voter1).claim())
+          .to.emit(voterReward, "Claim")
+        })
+      })
+
+      context("Failed claim", () => {
+        let voterReward = ethers.utils.parseEther('10');
+
+        it("should revert if voter has no rewards left to claim", async () => {
+          let rewards: any = [[voter1.address, voterReward], [voter2.address, voterReward]]
+          await collection.connect(sarcoDao).setRewards(rewards)
+
+          await collection.connect(voter1).claim()
+          // Try to call claim() again but will revert as balance of voter has been set to 0 after initial claim() call
+          await expect(collection.connect(voter1).claim()).to.be.revertedWith('InsufficientVoterBalance')
+        })
+      })
+    })
+
+    describe("daoWithdraw()", () => {
+      context("Successfully withdraw any excess rewards by DAO/Owner", () => {
+        let voterReward = ethers.utils.parseEther('10');
+
+        beforeEach(async () => {
+          let rewards: any = [[voter1.address, voterReward], [voter2.address, voterReward]]
+          await collection.connect(sarcoDao).setRewards(rewards)
+        });
+
+        it("should transfer remaining balance of the contract", async () => {
+          const withdrawableByDao = await collection.withdrawableByDao()
+
+          await collection.connect(sarcoDao).daoWithdraw()
+
+          // confirm the DAO/Owner receives only what is withdrawable, excluding what is claimable by voters
+          expect(await sarco.balanceOf(sarcoDao.address)).to.equal(withdrawableByDao)
```

Comment:
I would add one or two more `expect`s over here to verify that the contract's claimable rewards have been untouched